lib/features/context-pad/ContextPadProvider.js

```diff
@@ -192,14 +190,17 @@ ContextPadProvider.prototype.getContextPadEntries = function(element) {
       connect = this._connect,
       create = this._create,
       popupMenu = this._popupMenu,
-      rules = this._rules,
       autoPlace = this._autoPlace,
       translate = this._translate,
       appendPreview = this._appendPreview;
 
   var actions = {};
 
   if (element.type === 'label') {
+    if (this._isDeleteAllowed([ element ])) {
+      addDeleteAction();
```

Comment:
Slightly less magic, and clear on the side-effects we cause:

```suggestion
      assign(actions, deleteAction());
```

Reply:
Or, alternatively:

```suggestion
      addDeleteAction(actions);
```